src/anemoi/graphs/generate/icosahedral.py

```diff
@@ -5,50 +5,54 @@
 import networkx as nx
 import numpy as np
 import trimesh
-from sklearn.metrics.pairwise import haversine_distances
 from sklearn.neighbors import BallTree
 
+from anemoi.graphs.generate.masks import KNNAreaMaskBuilder
 from anemoi.graphs.generate.transforms import cartesian_to_latlon_rad
+from anemoi.graphs.generate.utils import get_coordinates_ordering
 
 
 def create_icosahedral_nodes(
-    resolutions: list[int],
+    resolution: int, aoi_mask_builder: KNNAreaMaskBuilder | None = None
```

Comment:
why is resolutions changing from list to int?

Reply:
In both hex & tri refinements, all nodes from one level are contained in the next refinement level. Before, we were getting the maximum of the list.

PS: The list of resolutions is still needed to define the scales at which to build the connections.